packages/core/test/oracle/Proposer.js

```diff
@@ -0,0 +1,182 @@
+const hre = require("hardhat");
+const { getContract, assertEventEmitted, web3 } = hre;
+const { runDefaultFixture, interfaceName, didContractThrow } = require("@uma/common");
+const { assert } = require("chai");
+
+const MockOracleCombined = getContract("MockOracleCombined");
+const MockOracleAncillary = getContract("MockOracleAncillary");
+const VotingToken = getContract("VotingToken");
+const Finder = getContract("Finder");
+const Governor = getContract("Governor");
+const Proposer = getContract("Proposer");
+const Timer = getContract("Timer");
+const IdentifierWhitelist = getContract("IdentifierWhitelist");
+const Store = getContract("Store");
+const { toWei, utf8ToHex } = web3.utils;
+
+describe("Proposer", function () {
+  let accounts;
+  let owner;
+  let submitter;
+  let rando;
+
+  let proposer;
+  let bond = toWei("100");
+
+  let mockOracle;
+  let votingToken;
+  let finder;
+  let governor;
+
+  before(async function () {
+    accounts = await web3.eth.getAccounts();
+    [owner, submitter, rando] = accounts;
+    await runDefaultFixture(hre);
```

Comment:
nit: where possible I think we should avoid these defaultFixtures and deploy all the contracts required by the tests _in_ the tests. we are trying to do this more and more with the new code.

Reply:
Why do you think we should avoid these?
I can give two reasons why I think it's a _good_ starting point:
- Less test code.
- More accurately reflects the deployed system in production. Put differently, you have to deliberately change things where you want the test to differ from our standard production deployment.

Thoughts?

Reply:
main downside from my experience is that it is slower to use this. This should not be the case with how hardhat snapshotting is meant to work but in my experience it is slower.

either way, I think in time we'd like to use _more_ of these kinds of fixtures. would be great to come up with a set that enable us to delete big chunks from before each blocks.

Reply:
I do try to avoid some of the costs by ensuring this only runs in the before and not the beforeEach, but I hear you. This does interact with a fair number of UMA contracts (store, governor, etc), so I think it makes sense to leave this using this same fixture, especially since the same fixture is used throughout the oracle suite.

It does illustrate that we need a more concrete plan for how to work with fixtures and test environments generally.

On a positive note: our tests are pretty quick nowadays!